contest/contest_info.py

```diff
@@ -108,17 +110,59 @@ def get_clarifications(user,contest):
         return reply_all | user_ask
     return reply_all
 
-def is_contestant(user,contest):
+def is_contestant(user, contest):
     if user.is_authenticated():
         contestant = Contestant.objects.filter(contest = contest, user = user)
         if len(contestant) >= 1:
             return True
     return False
 
-def can_ask(user,contest):
+#check if user can create new clarification in contest
+def can_ask(user, contest):
     user_is_contestant = is_contestant(user,contest)
     user_is_owner_coowner = has_contest_ownership(user,contest)
     return  user_is_contestant | user_is_owner_coowner
 
-def can_reply(user,contest):
+#check if user can reply clarification
+def can_reply(user, contest):
     return has_contest_ownership(user,contest)
+
+#check if user can edit contest
+def can_edit_contest(user, contest):
+    return user.has_admin_auth() or user_info.has_contest_ownership(user, contest)
+
+#check if user can create contest
+def can_create_contest(user):
+    return user.has_judge_auth()
+
+#check if user can delete contest
+def can_delete_contest(user):
+    return user.has_admin_auth() or (user == contest.owner)
+
+def can_register(user, contest):
+    #admin can't attend any contest
+    if not user_is_valid(user):
```

Comment:
not login user would use this function to judge too. But if adding that anotation, not login user can't call this function, which will cause problems.
BTW, since this kind of function won't affect db, calling this function by any user won't be a big trouble, in my opinion.